Gordon360/Services/RecIM/SportService.cs

```diff
@@ -34,6 +34,17 @@ public SportViewModel GetSportByID(int sportID)
             return _context.Sport.Find(sportID);
         }
 
+        public async Task<SportViewModel> DeleteSportAsync(int sportID)
+        {
+            var sport = _context.Sport.Find(sportID);
+            var activities = _context.Activity.Where(a => a.SportID == sportID);
+            foreach(var activity in activities)
+                activity.SportID = 0;
```

Comment:
This can be accomplished much more performantly and simply using the `ON DELETE SET DEFAULT` constraint in SQL Server, which just requires you to set `0` as the default value for Activity.SportID.

Reply:
Potentially we could do that, but I've been incredibly frustrated with "default" values when translated in EFCore as it considers it a nullable. Then casting becomes ugly everywhere else. But this would definitely be more safe overall if we're okay with having either a bunch of new extensions or a bunch of `??` on our model->viewmodel->model

Reply:
Hmm, I don't think I've encountered that with default values in EF Core before. That would be very frustrating.

Regardless, this is not a blocking issue. Maybe it's worth just adding an issue/todo to investigate this further somewhere down the line?

Reply:
I could make an issue for study or research.